TheThingsNetwork/lorawan-stack
# Network Server testing

#### Summary


References #960 
The US915 flow test will be added later and it will be largely based on this one, so decided to first bring the existing flow test in order.

#### Changes


- Do ~all~ *not do any* observability/metrics stuff in goroutines ~(no reason to block execution of critical functionality)~
- Use `networkserver.StartTest` in flow tests
- Add more event testing utilities
- Ensure NS tests does not panic on missed function calls
- Remove custom `ns.Close()` logic and rely exclusively on context
